src/dialogs/ShuffleDialog.java

```diff
@@ -83,19 +92,29 @@ public ShuffleDialog(ArrayManager ArrayManager, JFrame frame, UtilFrame utilFram
 
         bypassEvents = true;
         this.shuffleEditor.graph = ArrayManager.getShuffle();
+        jTextField1.setText(Double.toString(shuffleEditor.graph.sleepRatio));
         jList4.setListData(ArrayManager.getDistributionIDs());
         for(int i = 0; i < ArrayManager.getDistributions().length; i++) {
             if(ArrayManager.getDistribution().equals(ArrayManager.getDistributions()[i])) {
                 jList4.setSelectedIndex(i);
                 break;
             }
         }
-        jList1.setListData(ArrayManager.getDistributionIDs());
-        jList3.setListData(ArrayManager.getDistributionIDs());
+
+        distributions = Arrays.stream(ArrayManager.getDistributions())
+                              .filter(dist -> dist.getName() != "Custom")
+                              .collect(Collectors.toList());
+        distributionNames = distributions.stream()
+                                         .map(Distributions::getName)
+                                         .collect(Collectors.toList());
+        Object[] distributionNamesArray = distributionNames.toArray();
```

Comment:
why lists?

Reply:
Because that helps how they're used later.

Reply:
by being able to directly call `stream()`? (also, if my above comment holds, why not just directly `toArray()` the distribution names?)